ui/admin-portal/src/app/components/candidates/intake/avail-immediate/avail-immediate.component.ts

```diff
@@ -38,16 +42,44 @@ export class AvailImmediateComponent extends IntakeComponentBase implements OnIn
   }
 
   ngOnInit(): void {
+    const interested = this.computeInterested();
     this.form = this.fb.group({
       availDate: [this.candidateIntakeData?.availDate],
       availImmediate: [this.candidateIntakeData?.availImmediate],
       availImmediateJobOps: [this.candidateIntakeData?.availImmediateJobOps],
       availImmediateReason: [this.candidateIntakeData?.availImmediateReason],
       availImmediateNotes: [this.candidateIntakeData?.availImmediateNotes],
-      interested: []
+      interested: [interested]
     });
   }
 
+  /**
+   * Returns whether candidate is interested based on their status and their answer to
+   * availImmediate.
+   * @return yes or no - or null if unsure.
+   * @private
+   */
+  private computeInterested(): YesNo {
+    //Default is that the candidate has not responded to this question, or that they are unsure.
+    //Either way, we return null.
+    let interested = null;
+
+    //If the candidate's status is withdrawn, assume that they are not interested.
+    if (this.candidate.status == 'withdrawn') {
+      interested = YesNo.No;
+    } else {
+      //If the candidate's status is not that they have withdrawn then compute their status from
+      //any response they have made to the availImmediate question.
+      //If they have got as far as answering that question - even with a "NoResponse" response,
+      //we can assume that they said that they were interested.
+      if (this.candidateIntakeData?.availImmediate &&
+        this.candidateIntakeData?.availImmediate != YesNo.NoResponse) {
```

Comment:
The logic is more cautious than the comment suggests. The code only considers them "interested" if they actively responded with a Yes or No. It will not consider them interested if they actively responded with NoRsponse.

This goes against the sense of the comment - guessing it's maybe a typo?

Should the check simply be:

if (this.candidateIntakeData?.availImmediate) {

Then, any response to the availability question will be deemed to express interest, which I think is the intent here.

Reply:
Good pick up - but it is the comment that is wrong.

The code is the way it is supposed to be.